Tests/Tests/RouterSpec.swift

```diff
@@ -217,6 +260,48 @@ class RouterSpec: QuickSpec {
                 }
             }
 
+            context("when calling isDisplayingTooltip") {
+
+                let tooltipTargetView = UIView(frame: CGRect(x: 100, y: 100, width: 10, height: 10))
+                let tooltip = TestHelpers.generateTooltip(id: "test")
+                let imageBlob: Data! = UIImage(named: "test-image", in: .unitTests, with: nil)?.pngData()
+
+                beforeEach {
+                    tooltipTargetView.accessibilityIdentifier = TooltipViewIdentifierMock
+                    window.addSubview(tooltipTargetView)
+                }
+
+                afterEach {
+                    tooltipTargetView.removeFromSuperview()
+                    window.findTooltipView()?.removeFromSuperview()
+                }
+
+                it("will return true when tooltip displayed") {
+                    router.displayTooltip(tooltip, targetView: tooltipTargetView,
+                                          identifier: TooltipViewIdentifierMock,
+                                          imageBlob: imageBlob,
+                                          becameVisibleHandler: { _ in },
+                                          confirmation: true,
+                                          completion: { _ in })
+                    expect(window.findTooltipView()).toEventuallyNot(beNil())
+                    expect(router.isDisplayingTooltip(with: TooltipViewIdentifierMock)).to(beTrue())
+                }
+
+                it("will return false when tooltip is not displayed") {
+                    expect(router.isDisplayingTooltip(with: "test")).to(beFalse())
+                }
+
+                it("will keep return Bool value when calling the from background thread") {
```

Comment:
Is this test checking if there's no thread lock?
In that case you can just say "will return a value ..." in the beginning

Reply:
This case is checking whether the function will work properly when the function is called in a background thread.
It will cover this line.
<img width="609" alt="image" src="https://user-images.githubusercontent.com/121004972/236720403-70f663b6-0db4-4391-af14-c938974fdf9e.png">